fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/If.java

```diff
@@ -90,15 +84,6 @@ public class If extends ScalarFunction
                     .args(BooleanType.INSTANCE, StringType.INSTANCE, StringType.INSTANCE)
     );
 
-    private final Supplier<DataType> widerType = Suppliers.memoize(() -> {
-        List<AbstractDataType> argumentsTypes = getSignature().argumentsTypes;
-        Type assignmentCompatibleType = ScalarType.getAssignmentCompatibleType(
-                argumentsTypes.get(1).toCatalogDataType(),
-                argumentsTypes.get(2).toCatalogDataType(),
-                true);
-        return DataType.fromCatalogType(assignmentCompatibleType);
-    });
-
     /**
```

Comment:
wider type is nessesary, if there is some bug, you should fix it, but not delete this widerType

Reply:
fixed